Added doc for TaxonomyNoEntries & TaxonomySubtree (#3082)
ezsystems/developer-documentation: docs/search/criteria_reference/taxonomy_no_entries.md

---
description: TaxonomyNoEntries Search Criterion
---

# TaxonomyNoEntries Criterion

> The [`TaxonomyNoEntries` Search Criterion](https://example.com/api/php_api/php_api_reference/classes/Ibexa-Contracts-Taxonomy-Search-Query-Criterion-TaxonomyNoEntries.html) searches for content that has no entries assigned from the specified [taxonomy](taxonomy.md).

Author: The example.com is a TMP change that must be removed before merge - the criterion is not part of the PHP API reference until it's released

> The [`TaxonomyNoEntries` Search Criterion](https://example.com/api/php_api/php_api_reference/classes/Ibexa-Contracts-Taxonomy-Search-Query-Criterion-TaxonomyNoEntries.html) searches for content that has no entries assigned from the specified [taxonomy](taxonomy.md).

Reviewer: ```suggestion
The [`TaxonomyNoEntries`](https://example.com/api/php_api/php_api_reference/classes/Ibexa-Contracts-Taxonomy-Search-Query-Criterion-TaxonomyNoEntries.html) Search Criterion searches for content that has no entries assigned from the specified [taxonomy](taxonomy.md).
```

> The [`TaxonomyNoEntries`](https://example.com/api/php_api/php_api_reference/classes/Ibexa-Contracts-Taxonomy-Search-Query-Criterion-TaxonomyNoEntries.html) Search Criterion searches for content that has no entries assigned from the specified [taxonomy](taxonomy.md).

Reviewer 2: Don't forget to update this before merge:
```suggestion
The [`TaxonomyNoEntries`](/api/php_api/php_api_reference/classes/Ibexa-Contracts-Taxonomy-Search-Query-Criterion-TaxonomyNoEntries.html) Search Criterion searches for content that has no entries assigned from the specified [taxonomy](taxonomy.md).
```

Use it when you need to find content items to which no taxonomy entries have been assigned (for example, articles without tags).

> It's available for all supported search engines and in [repository filtering](search_api.md#repository-filtering)

Author: ```suggestion
It's available for all supported search engines and in [repository filtering](search_api.md#repository-filtering).
```

> It is available for all supported search engines and in [repository filtering](search_api.md#repository-filtering)

Reviewer: ```suggestion
It's available for all supported search engines and in [repository filtering](search_api.md#repository-filtering)
```

> It is available for all supported search engines and in [repository filtering](search_api.md#repository-filtering)

Reviewer 3: ```suggestion
It's available for all supported search engines and in [repository filtering](search_api.md#repository-filtering).
```

## Arguments

- `taxonomy` - `string` representing the identifier of the taxonomy (for example, `tags` or `categories`)

## Example

### PHP

The following example searches for articles that have no entries assigned in the `tags` taxonomy:

```php hl_lines="11-14"
[[= include_file('code_samples/search/content/taxonomy_no_entries_criterion.php') =]]
```

> The criteria limit the results to content matching all of the conditions listed below:

Reviewer 3: ```suggestion
The criteria limit the results to content that matches all of the conditions listed below:
```
